azurelinuxagent/common/cgroups.py

```diff
@@ -796,3 +778,32 @@ def set_memory_limit(self, limit=None, unit='megabytes'):
             fileutil.write_file(memory_limit_file, '{0}\n'.format(value))
         else:
             raise CGroupsException("Memory hierarchy not available in this cgroup")
+
+
+class CGroupsLimits(object):
+    def __init__(self, cgroup_name, threshold=None):
+        if not cgroup_name or cgroup_name == "":
+            cgroup_name = "Agents+Extensions"
+
+        self.cpu_limit = threshold["cpu"] if threshold and "cpu" in threshold else self.get_default_cpu_limits(
+            cgroup_name)
+        self.memory_limit = threshold["memory"] if threshold and "memory" in threshold else self.get_default_memory_limits(
```

Comment:
Suggestion (feel free to ignore me): I'm a big fan of DRY. These two lines are mostly repeated; if you added a third limit, it would be mostly a repeat of these. Consider refactoring to extract the common code:

```
    def _get_value_or_default(threshold, limit, compute_default):
        return threshold[limit] if threshold and limit in threshold else compute_default(self, self.cgroup_name)
```

That makes these two lines look like:
```
    self.cpu_limit = self._get_value_or_default(threshold, "cpu", CGroupsLimits.get_default_cpu_limits)
    self.memory_limit = self._get_value_or_default(threshold, "memory", CGroupsLimits.get_default_memory_limits)
```

As I said, this is just a suggestion.